invalidate.cloudfront.js

```diff
@@ -0,0 +1,52 @@
+const exec = require('child_process').exec;
+const colors = require('colors');
+const AWS = require('aws-sdk');
+
+const cloudfront = new AWS.CloudFront();
+const pathToInvalidate = `/searchui/v${process.env.PACKAGE_JSON_VERSION}/*`;
+
+const shouldDoInvalidation = () => {
+  if (!process.env.TRAVIS) {
+    return false;
+  }
+  if (!process.env.TRAVIS_TAG) {
+    return false;
+  }
+  if (process.env.TRAVIS_TAG.indexOf('beta') != -1) {
+    return false;
+  }
+  return true;
+};
+
+if (shouldDoInvalidation()) {
+  const invalidationRequest = cloudfront.createInvalidation({
+    DistributionId: process.env.AWS_CLOUDFRONT_DISTRIBUTION_ID,
+    InvalidationBatch: {
+      CallerReference: new Date().getTime().toString(),
+      Paths: {
+        Quantity: 1,
+        Items: [pathToInvalidate]
+      }
+    }
+  });
+
+  invalidationRequest.send((error, success) => {
+    if (error) {
+      console.log(colors.red('ERROR WHILE INVALIDATING RESSOURCES ON CLOUDFRONT'));
+      console.log(colors.red('*************'));
+      console.log(colors.red(error.message));
+      console.log(colors.red('*************'));
+      exec('travis_terminate 1');
+      throw error;
```

Comment:
Is it necessary to throw this error even though you are terminating the process on line 39? Are we catching or relying on node.js to catch? If the latter, wouldn't that crash the server?

Edit: I guess this will only crash the travis instance which isn't a big deal. But still, why do it if we terminate on the line before?

Reply:
True, the throw is not 100% needed.

It's going to print the error in the console though, which can be useful to debug.

The travis terminate will cause the travis build to fail. The server will not "crash".